src/ddmd/root/filename.d

```diff
@@ -680,20 +677,36 @@ nothrow:
         }
         else version (Windows)
         {
+            import core.sys.windows.winbase: GetFullPathNameW;
+
+            wchar[1024] wpathBuf;
+            const wpath = name.toWStringz(wpathBuf);
+
             /* Apparently, there is no good way to do this on Windows.
              * GetFullPathName isn't it, but use it anyway.
              */
-            DWORD result = GetFullPathNameA(name, 0, null, null);
-            if (result)
+            DWORD length16 = GetFullPathNameW(&wpath[0], 0, null, null);
+            if (length16)
             {
-                char* buf = cast(char*)malloc(result);
-                result = GetFullPathNameA(name, result, buf, null);
-                if (result == 0)
+                auto buf = cast(wchar*)malloc(length16);
+                length16 = GetFullPathNameW(&wpath[0], length16, buf, null);
+                if (length16 == 0)
                 {
                     .free(buf);
                     return null;
                 }
-                return buf;
+
+                // allocate enough space for a UTF8 encoding of buf
+                const length8 = length16 * 4 + 1;
```

Comment:
should be `3` not `4`, worst case for translating a code point from `utf-16` to `utf-8` is 3 `utf-8` code units for 1 `utf-16` code unit, which happens for code units `0x000800 - 0x00FFFF`

Reply:
Fixed.